### PR TITLE
Update Gateway templates and parameters

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

File: dev/host_vars/apim-gateway_1.yml [hunk 1]
```diff
@@ -18,28 +18,26 @@
...
+  - { src: 'carbon-home/repository/conf/carbon.xml.j2',
+      dest: '{{ carbon_home }}/repository/conf/carbon.xml' }
+  - { src: 'carbon-home/repository/conf/api-manager.xml.j2',
+      dest: '{{ carbon_home }}/repository/conf/api-manager.xml' }
+  - { src: 'carbon-home/repository/conf/axis2/axis2.xml.j2',
+      dest: '{{ carbon_home }}/repository/conf/axis2/axis2.xml' }
+  - { src: 'carbon-home/repository/conf/tomcat/catalina-server.xml.j2',
+      dest: '{{ carbon_home }}/repository/conf/tomcat/catalina-server.xml' }
+  - { src: 'carbon-home/bin/wso2server.sh.j2',
+      dest: '{{ carbon_home }}/bin/wso2server.sh' }
```
Comment (VimukthiPerera) [inline dev/host_vars/apim-gateway_1.yml]: Missing empty line at EOF

Comment (Savidude) [reply dev/host_vars/apim-gateway_1.yml]: Fixed https://github.com/wso2/ansible-apim/pull/48/commits/65aa90071439321aa547410c7995aa0e4003b2b0

File: roles/apim-gateway/templates/carbon-home/repository/conf/tomcat/catalina-server.xml.j2 [hunk 2]
```diff
@@ -75,8 +75,8 @@
                    compressionMinSize="2048"
                    noCompressionUserAgents="gozilla, traviata"
                    compressableMimeType="text/html,text/javascript,application/x-javascript,application/javascript,application/xml,text/css,application/xslt+xml,text/xsl,image/gif,image/jpg,image/jpeg"
-                   keystoreFile="${carbon.home}/repository/resources/security/wso2carbon.jks"
-                   keystorePass="wso2carbon"
+                   keystoreFile="${carbon.home}/{{ key_store_location }}"
```
Comment (VimukthiPerera) [inline roles/apim-gateway/templates/carbon-home/repository/conf/tomcat/catalina-server.xml.j2]: The whole path should be parameterized. If the user adds the key file to an outside location, `${carbon.home}` path prefix will give you an error.

File: roles/apim-gateway/templates/carbon-home/repository/conf/carbon.xml.j2 [hunk 3]
```diff
@@ -407,15 +407,15 @@
         -->
         <InternalKeyStore>
             <!-- Keystore file location-->
-            <Location>{{ internal_keystore_location }}</Location>
+            <Location>${carbon.home}/repository/resources/security/wso2carbon.jks</Location>
```
Comment (VimukthiPerera) [inline roles/apim-gateway/templates/carbon-home/repository/conf/carbon.xml.j2]: Please do not remove the parameterization here. Keep it common for all the profiles as discussed.

**Pull request merged**